### ISSUE TITLE
remote_path is not respected when become is true for script module

### ISSUE THREAD

**Issue by gjelu** [ansible/ansible#22203]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
sccript module

##### ANSIBLE VERSION
ansible 2.1.2.0
  config file = /home/lsc8100/.ansible.cfg
  configured module search path = Default w/o overrides

##### CONFIGURATION
```ini
[defaults]
remote_tmp = /var/opt/app/.ansible/tmp/
host_key_checking = False
timeout = 30
retry_files_enabled = False
retry_files_save_path = ~/.ansible-retry
hash_behaviour=merge
```
##### OS / ENVIRONMENT
Ansible server: RedHat 6.8
target host: RedHat 7.3 but /tmp is mounted with noexec option

##### SUMMARY
when a playbook is started as become:true and become_user, a script from a role is copied to /tmp/ instead of the remote_tmp and executed from this location. 

As the /tmp is mounted as noexec, we cannot execute the script. 

##### STEPS TO REPRODUCE
```yaml
playbook:
- hosts: all
  become: true
  become_user: jenkins
  gather_facts: no
  roles:
   - role: myrole
```
the role 'myrole' has the following myrole/tasks/main.yaml
```yaml
 - name: execute test
   script: test.sh
   register: cmdtest
```
The dummy script test.sh is located in 'myrole/files'

##### EXPECTED RESULTS

The script test should be copied in the remote_tmp location being, /var/opt/app/.ansible/tmp/
and should be executed from here. 

##### ACTUAL RESULTS

The script is copied in /tmp. 
```
fatal: [inetyiaplv003]: FAILED! => {"changed": true, "failed": true, "rc": 126, "stderr": "", "stdout": "/bin/sh: /tmp/ansible-tmp-1488470170.18-148393376669627/test.sh: Permission denied\r\n", "stdout_lines": ["/bin/sh: /tmp/ansible-tmp-1488470170.18-148393376669627/test.sh: Permission denied"]}
```
When we remove the become and become_user from the playbook, the playbook is executed fine.

**Comment by bcoca** [comment]
The problem here is that `remote_tmp` is ignored if you are using `become` for a non root user, it falls back to 'system' temps so only `/var/tmp` or `/tmp` are allowed. We do plan on making this list configurable in the future though.

**Comment by jcrisologo** [comment]
I am also trying to find a workaround to this problem due to a STIG requirement that `/tmp` be noexec (questionably useful but them's the breaks, yeah?).  I have tried on Ansible 2.3.2 and 2.1.2.  My error is slightly different from OP's, and I can't get Ansible to work even after removing `become`.
```
[jcrisologo@master ansible]$ ansible -kvi inventory/test -m ping -u ansible slave
SSH password:
slave | FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Shared connection to 192.168.33.34 closed.\r\n",
    "module_stdout": "/usr/bin/python: can't open file '/tmp/ansible_WmOXgr/ansible_module_ping.py': [Errno 13] Permission denied\r\n",
    "msg": "MODULE FAILURE",
    "rc": 0
}
```
Is it the case that Ansible will always attempt to run executables out of the system tmpdir, even without `become`?
I ask because this snippet from `module_common.py` appears to run no matter what, not even pipeline mode prevents an executable from being dropped in `/tmp`
```python
# There's a race condition with the controller removing the
# remote_tmpdir and this module executing under async.  So we cannot
# store this in remote_tmpdir (use system tempdir instead)
temp_path = tempfile.mkdtemp(prefix='ansible_')
```

**Comment by alikins** [comment]
See also #31022

**Comment by bcoca** [comment]
fixed via #31677 which made system temp dirs configurable
